src/encoded/genomic_data_service.py

```diff
@@ -0,0 +1,136 @@
+import requests
+
+REGISTRY_DATA_SERVICE = 'genomic_data_service'
+RNA_GET_FILTERS = ['assayType', 'annotation']
+RNA_GET_ENDPOINT = '/expressions/bytes'
+
+# react component orders columns by "the position" in the hash map
+RNA_GET_COLUMNS = {
+    'featureID': {'title': 'Feature ID'},
+    'tpm': {'title': 'Counts (TPM)'},
+    'fpkm': {'title': 'Counts (FPKM)'},
+    'assayType': {'title': 'Assay (RNA SubType)'},
+    'libraryPrepProtocol': {'title': 'Experiment'},
+    'expressionID': {'title': 'File'},
+    'annotation': {'title': 'Genome Annotation'}
+}
+
+class GenomicDataService():
+    # default search value is a temporary feature for the current alpha UI client
+    def __init__(self, registry, request, default_search='ENSG00000088320.3'):
+        self.path = registry.settings.get(REGISTRY_DATA_SERVICE)
+        self.request = request
+        self.default_search = default_search
+
+        self.parse_params()
+
+
+    def parse_params(self):
+        params = self.request.params
+
+        self.genes   = params.get('genes')
+        self.units   = params.get('units', 'tpm')
+        self.page    = params.get('page', 1)
+        self.sort    = params.get('sort')
+        self.columns = RNA_GET_COLUMNS.copy()
+        self.columns.pop('tpm' if self.units == 'fpkm' else 'fpkm')
+
+        if self.sort:
+            desc = self.sort[0] == '-'
+            self.sort_by = self.sort[1:] if desc else self.sort
+            self.sort_order = 'desc' if desc else 'asc'
+
+        self.filter_params = {}
+        for filter_ in RNA_GET_FILTERS:
+            if params.get(filter_):
+                self.filter_params[filter_] = params.get(filter_)
+
+        self.filters = [{'field': 'type', 'term': 'Rna Get', 'remove': '/rnaget'}]
+
+        for filter_ in self.filter_params:
```

Comment:
Suggestion for the future. I've been using https://github.com/ENCODE-DCC/snovault/blob/9fb33ae20ac5b7de775d0455838b2a6b3906be60/src/snovault/elasticsearch/searches/parsers.py#L441 for most of these common param parsing/querystring tasks. Would have to look at interface to see how it works but maybe some use here.

Reply:
oh good to know! I will take a look